### PR TITLE
Update to catch up with DEOptim

### PR DESCRIPTION
**Pull request by Ecthilisyk**
Add functionality to randomize cross-over probability and step-size coefficient in each mutation

### CODE REVIEW

File: src/deoptim.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ void devol(double VTR, double f_weight, double fcross, int i_bs_flag,
            arma::mat    & ta_popP, arma::mat    & ta_oldP, arma::mat    & ta_newP, arma::colvec & t_bestP, 
            arma::colvec & ta_popC, arma::colvec & ta_oldC, arma::colvec & ta_newC, double       & t_bestC,      
            arma::colvec & t_bestitP, arma::colvec & t_tmpP, arma::mat & d_pop, Rcpp::List & d_storepop, 
-           arma::mat & d_bestmemit, arma::colvec & d_bestvalit, int & i_iterations, double i_pPct, long & l_nfeval,
+           arma::mat & d_bestmemit, arma::colvec & d_bestvalit, int & i_iterations, double i_pPct, double d_c, long & l_nfeval,
```
Comment (eddelbuettel) [inline src/deoptim.cpp:21]: If you change the interface, then `RcppExports.{cpp,R}` change. You must run roxygen on it, which RStudio may do for you.

Comment (Ecthilisyk) [inline src/deoptim.cpp:21]: hmm, this is supposed to be part of a **list** variable in 'RcppExport.{cpp, R}'. Last time when I added `d_reltol` and `i_steptol` the RcppExport files were changed

Comment (eddelbuettel) [inline src/deoptim.cpp:21]: Look at the files included in the commit (as listed on this page). They weren't. That is likely the compilation failure.

Comment (eddelbuettel) [inline src/deoptim.cpp:21]: Oh -- I take that back. The change is only in the control structure so the interface to R did not change.  No `RcppExports.{cpp,R}`  update needed -- I was just plain wrong.

File: src/devol.cpp [hunk 2]
```diff
@@ -103,6 +107,18 @@ void devol(double VTR, double f_weight, double f_cross, int i_bs_flag,
...
+            if(d_c>0){
+              f_cross = Rcpp::as<double>(rnorm(1, mean_cross, 0.1));
+              f_cross = f_cross > 1.0 ? 1 : f_cross;
+              f_cross = f_cross < 0.0 ? 0 : f_cross;
+              do{
+                f_weight = Rcpp::as<double>(rcauchy(1, mean_weight, 0.1));
+                f_weight = f_weight > 1.0 ? 1.0 : f_weight;
+              } while (f_weight <= 0.0);
+            }
+
```
Comment (Ecthilisyk) [inline src/devol.cpp:121]: I think I know the reason. `f_cross` and `f_weight` are passed in as const parameters, the code above attempts to change them. let me make the change

Comment (eddelbuettel) [inline src/devol.cpp:121]: :+1:

Continuous Integration and unit tests are very valuable in catching these things,

**Comment by eddelbuettel** [comment]
As you filed another one I will close this one.

**Pull request closed without merging**